src/ReactNativeZoomableView.tsx

```diff
@@ -71,233 +87,217 @@ class ReactNativeZoomableView extends Component<
           evt,
           gestureState,
-          this._getZoomableViewEventObject()
+          _getZoomableViewEventObject()
         );
       },
       onPanResponderTerminationRequest: (evt, gestureState) =>
-        !!this.props.onPanResponderTerminationRequest?.(
+        !!props.onPanResponderTerminationRequest?.(
           evt,
           gestureState,
-          this._getZoomableViewEventObject()
+          _getZoomableViewEventObject()
         ),
       // Defaults to true to prevent parent components, such as React Navigation's tab view, from taking over as responder.
       onShouldBlockNativeResponder: (evt, gestureState) =>
-        this.props.onShouldBlockNativeResponder?.(
+        props.onShouldBlockNativeResponder?.(
           evt,
           gestureState,
-          this._getZoomableViewEventObject()
+          _getZoomableViewEventObject()
         ) ?? true,
       onStartShouldSetPanResponderCapture: (evt, gestureState) =>
-        !!this.props.onStartShouldSetPanResponderCapture?.(evt, gestureState),
+        !!props.onStartShouldSetPanResponderCapture?.(evt, gestureState),
       onMoveShouldSetPanResponderCapture: (evt, gestureState) =>
-        !!this.props.onMoveShouldSetPanResponderCapture?.(evt, gestureState),
+        !!props.onMoveShouldSetPanResponderCapture?.(evt, gestureState),
     });
 
-    this.zoomSubjectWrapperRef = createRef<View>();
-
-    if (this.props.zoomAnimatedValue)
-      this.zoomAnim = this.props.zoomAnimatedValue;
-    if (this.props.panAnimatedValueXY)
-      this.panAnim = this.props.panAnimatedValueXY;
+    if (props.zoomAnimatedValue) zoomAnim.current = props.zoomAnimatedValue;
+    if (props.panAnimatedValueXY) panAnim.current = props.panAnimatedValueXY;
 
-    if (this.props.initialZoom) this.zoomLevel = this.props.initialZoom;
-    if (this.props.initialOffsetX != null)
-      this.offsetX = this.props.initialOffsetX;
-    if (this.props.initialOffsetY != null)
-      this.offsetY = this.props.initialOffsetY;
+    if (props.initialZoom) zoomLevel.current = props.initialZoom;
+    if (props.initialOffsetX != null) offsetX.current = props.initialOffsetX;
+    if (props.initialOffsetY != null) offsetY.current = props.initialOffsetY;
 
-    this.panAnim.setValue({ x: this.offsetX, y: this.offsetY });
-    this.zoomAnim.setValue(this.zoomLevel);
-    this.panAnim.addListener(({ x, y }) => {
-      this.offsetX = x;
-      this.offsetY = y;
+    panAnim.current.setValue({ x: offsetX.current, y: offsetY.current });
+    zoomAnim.current.setValue(zoomLevel.current);
+    panAnim.current.addListener(({ x, y }) => {
+      offsetX.current = x;
+      offsetY.current = y;
     });
-    this.zoomAnim.addListener(({ value }) => {
-      this.zoomLevel = value;
+    zoomAnim.current.addListener(({ value }) => {
+      zoomLevel.current = value;
     });
-
-    this.lastGestureTouchDistance = 150;
-
-    this.gestureType = null;
-  }
-
-  componentDidUpdate(
```

Comment:
this function is now split into 3 layout effects, each check for a specific set of changes